<a href="https://colab.research.google.com/github/sagar2582/World-Bank-Binary-classifier/blob/main/World_Bank_Binary_Classifier.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from mpl_toolkits.mplot3d import Axes3D
from sklearn.preprocessing import StandardScaler
import matplotlib.pyplot as plt # plotting
import numpy as np # linear algebra
import os # accessing directory structure
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

In [7]:
# Connecting to Drive to import the dataset

from google.colab import drive
drive.mount('/content/drive/')

Mounted at /content/drive/


In [8]:
df = pd.read_csv('/content/drive/My Drive/Taiyo/world_bank_dataset.csv')

In [9]:
# First look
df.head()

id                    regionname  \
0  P151169  Middle East and North Africa   
1  P154943         East Asia and Pacific   
2  P161382         East Asia and Pacific   
3  P163922         East Asia and Pacific   
4  P169279         East Asia and Pacific   

                                         countryname prodline  \
0              Kingdom of Morocco;Kingdom of Morocco       PE   
1                  Kingdom of Tonga;Kingdom of Tonga       PE   
2  Republic of the Marshall Islands;Republic of t...       PE   
3  Federated States of Micronesia;Federated State...       PE   
4                                        Samoa;Samoa       PE   

                   lendinginstr lendinginstrtype envassesmentcategorycode  \
0  Investment Project Financing               IN                        B   
1  Investment Project Financing               IN                        C   
2  Investment Project Financing               IN                        B   
3  Investment Project Financing               IN                        B   
4  Investment Project Financing               IN                      NaN   

  supplementprojectflg productlinetype projectstatusdisplay  ... mjtheme3name  \
0                    N               L               Active  ...          NaN   
1                    N               L               Active  ...          NaN   
2                    N               L               Active  ...          NaN   
3                    N               L               Active  ...          NaN   
4                    Y               L               Active  ...          NaN   

  mjtheme4name mjtheme5name location GeoLocID GeoLocName Latitude Longitude  \
0          NaN          NaN      NaN      NaN        NaN      NaN       NaN   
1          NaN          NaN      NaN      NaN        NaN      NaN       NaN   
2          NaN          NaN      NaN      NaN        NaN      NaN       NaN   
3          NaN          NaN      NaN      NaN        NaN      NaN       NaN   
4          NaN          NaN      NaN      NaN        NaN      NaN       NaN   

  Country Unnamed: 56  
0     NaN         NaN  
1     NaN         NaN  
2     NaN         NaN  
3     NaN         NaN  
4     NaN         NaN  

[5 rows x 57 columns]

In [10]:
df.tail()

id                   regionname  \
18836  P006578  Latin America and Caribbean   
18837  P037451      Europe and Central Asia   
18838  P037362      Europe and Central Asia   
18839  P037452      Europe and Central Asia   
18840  P037383      Europe and Central Asia   

                                             countryname prodline  \
18836                Republic of Chile;Republic of Chile       PE   
18837  Grand Duchy of Luxembourg;Grand Duchy of Luxem...       PE   
18838              Kingdom of Denmark;Kingdom of Denmark       PE   
18839  Kingdom of the Netherlands;Kingdom of the Neth...       PE   
18840                    French Republic;French Republic       PE   

                     lendinginstr lendinginstrtype envassesmentcategorycode  \
18836    Specific Investment Loan               IN                      NaN   
18837  Structural Adjustment Loan               AD                      NaN   
18838  Structural Adjustment Loan               AD                      NaN   
18839  Structural Adjustment Loan               AD                      NaN   
18840  Structural Adjustment Loan               AD                      NaN   

      supplementprojectflg productlinetype projectstatusdisplay  ...  \
18836                    N               L               Closed  ...   
18837                    N               L               Closed  ...   
18838                    N               L               Closed  ...   
18839                    N               L               Closed  ...   
18840                    N               L               Closed  ...   

      mjtheme3name mjtheme4name mjtheme5name location GeoLocID GeoLocName  \
18836          NaN          NaN          NaN      NaN      NaN        NaN   
18837          NaN          NaN          NaN      NaN      NaN        NaN   
18838          NaN          NaN          NaN      NaN      NaN        NaN   
18839          NaN          NaN          NaN      NaN      NaN        NaN   
18840          NaN          NaN          NaN      NaN      NaN        NaN   

      Latitude Longitude Country Unnamed: 56  
18836      NaN       NaN     NaN         NaN  
18837      NaN       NaN     NaN         NaN  
18838      NaN       NaN     NaN         NaN  
18839      NaN       NaN     NaN         NaN  
18840      NaN       NaN     NaN         NaN  

[5 rows x 57 columns]

In [13]:
df.shape

(18841, 57)

# Dataset Information
The shape of our dataset is (18841, 57)

id --> unique id-number for every project i.e., 18841 unique id's

regionname --> points towards
